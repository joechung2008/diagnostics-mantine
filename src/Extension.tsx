import { Text } from "@mantine/core";
import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="extension-root grow">
      <Text component="h2" size="xl" fw="bold">
        {extensionName}
      </Text>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
