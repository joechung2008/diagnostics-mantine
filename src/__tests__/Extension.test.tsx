import { describe, expect, it } from "vitest";
import Extension from "../Extension";
import { render } from "./test-utils";

describe("Extension", () => {
  it("should render the extension name", () => {
    const props: ExtensionInfo = { extensionName: "test" };
    const { container } = render(<Extension {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render Configuration if config is provided", () => {
    const props: ExtensionInfo = {
      extensionName: "test",
      config: { key: "value" },
    };
    const { container } = render(<Extension {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render StageDefinition if stageDefinition is provided", () => {
    const props: ExtensionInfo = {
      extensionName: "test",
      stageDefinition: { stage: ["step1"] },
    };
    const { container } = render(<Extension {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
