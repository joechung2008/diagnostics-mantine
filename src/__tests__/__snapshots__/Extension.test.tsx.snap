// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extension > should render Configuration if config is provided 1`] = `
<style
  data-mantine-styles="true"
>
  




</style>
`;

exports[`Extension > should render StageDefinition if stageDefinition is provided 1`] = `
<style
  data-mantine-styles="true"
>
  




</style>
`;

exports[`Extension > should render the extension name 1`] = `
<style
  data-mantine-styles="true"
>
  




</style>
`;
